trio/_core/_thread_cache.py

```diff
@@ -93,7 +93,7 @@ def __init__(self):
         self._idle_workers = {}
         self._cache_lock = Lock()
 
-    def start_thread_soon(self, deliver, fn):
+    def start_thread_soon(self, fn, deliver):
```

Comment:
For what it's worth, yes, this new order seems more intuitive to me, as deliver is called after fn